### PR TITLE
Feature More support for Python 3 and mixed Python 2, 2+3 and 3

### PR DESCRIPTION
**Pull request by bennygomspace**
Stuff relevant primary during migration to Python 3 added

* Added checkers:
** pylint-py3k (Command: pylint --py3k)
** pylint3

* Autodection if a file shall be checked with a subset of the specifed checkers:
** Python 2 compatible checkers
** Python 3 compatible checkers
** Both Python 2 and 3 compatible checkers
There will never be checked with checkers not specified via arg --checkers or in CHECKERS=... in config file.

Autodection is only active if enabled via arg --py23-autodetect or PY23_AUTODETECT = 1 in config file

### CODE REVIEW

File: bin/pycheckers.py [hunk 1]
```diff
@@ -535,6 +537,30 @@ def process_returncode(self, returncode):
         return not (returncode & 1 or returncode & 32)
 
 
+class PylintPy3kRunner(PylintRunner):
+    """Run pylint --py3k, producing flymake readable output.
```
Comment (msherry) [inline bin/pycheckers.py]: I know it's copied from existing checkers (that I need to fix!), but can we change "flymake" to "flycheck"?

File: bin/pycheckers.py [hunk 2]
```diff
@@ -535,6 +537,30 @@ def process_returncode(self, returncode):
...
+    """
+
+    @property
+    def name(self):
+        # type: () -> str
+        return 'pylint-py3k'
+
+
+class Pylint3Runner(PylintRunner):
+    """Run pylint3, producing flymake readable output.
```
Comment (msherry) [inline bin/pycheckers.py]: Same here, `s/flymake/flycheck/`.

File: bin/pycheckers.py [hunk 3]
```diff
@@ -753,11 +787,63 @@ def update_options_locally(options):
     return options
 
 
+def py2_py3_enable(py23_autodetect, source_file_path):
+    # type: (str) -> bool, bool
+    '''
+    Check if first line says python/python2/python3 => Activate py2/3 checkers
+
+    Can be overruled via a line: # py:<version>
```
Comment (msherry) [inline bin/pycheckers.py]: Interesting! Is this a common convention, or just something you decided?

File: bin/pycheckers.py [hunk 4]
```diff
@@ -753,11 +787,63 @@ def update_options_locally(options):
...
+    If <version> contains 2 -> then set py2 (1th return value)
```
Comment (msherry) [inline bin/pycheckers.py]: Nit: "1st" instead of "1th".

File: bin/pycheckers.py [hunk 5]
```diff
@@ -753,11 +787,63 @@ def update_options_locally(options):
     return options
 
 
+def py2_py3_enable(py23_autodetect, source_file_path):
+    # type: (str) -> bool, bool
```
Comment (msherry) [inline bin/pycheckers.py]: This type signature looks wrong -- there are two parameters, but only one input type.

**Comment by sorennohr** [comment]
#####################################
Subject: Review of flycheck-pycheckers pull #25
Hi Marc

https://github.com/msherry/flycheck-pycheckers/pull/25

Unfortunate Benny can't make the changes.
If you can make the changes, it will be great!

It's fine with us to:
- s/flymake/flycheck/g (twice)
- s/1th/1st/
- Auto-detection: See below
- Add 1st input bool:
  # type: (bool, str) -> bool, bool

Auto-detection
1st line
The auto detection if the first line is a simple matching for both
#!/path/to/pythonX and #!/usr/bin/env pythonX - any other pythonX
where X can be empty, 2 or 3.
The check is made simple, so bad #! formatted lines are recognized.

# py:<version>
The # py:<version> was not found elsewhere, actually we have not found any checking-help for python2-> 2+3 or 3 migration. (2to3 and futurize can modify the code but not show mistakes (well)).

The background was:
Space after # to let checkers be happy (some rules claims if space after # is missing)
py: Just a short identifier
<version> could be stricter checking, but a simple search for 2 and 3

**Pull request closed without merging**